hazardous/data/_competing_weibull.py

```diff
@@ -19,29 +21,118 @@
+            df_censoring_params,
+            shape_default=(1.0, 1.0),
+            scale_default=(1, 10 * censoring_relative_scale),
+            event=0,
+        )
+        scale_censoring = df_censoring_params["scale_0"].values * y["duration"].mean()
+        shape_censoring = df_censoring_params["shape_0"].values
+    censoring = weibull_min.rvs(
+        shape_censoring, scale=scale_censoring, size=y.shape[0], random_state=rng
+    )
+    y_censored = y.copy()
+    y_censored["event"] = np.where(y["duration"] < censoring, y_censored["event"], 0)
+    y_censored["duration"] = np.minimum(y_censored["duration"], censoring)
+    return y_censored
 
 
-def make_synthetic_competing_weibull(
+def compute_shape_and_scale(
+    df_features,
+    features_rate=0.2,
     n_events=3,
-    n_samples=3000,
-    return_X_y=False,
+    degree_interaction=2,
+    shape_ranges=DEFAULT_SHAPE_RANGES,
+    scale_ranges=DEFAULT_SCALE_RANGES,
+    random_state=0,
+):
+    rng = np.random.RandomState(random_state)
+    # Adding interactions between features
+    df_poly_features = PolynomialFeatures(
+        degree=degree_interaction, interaction_only=True, include_bias=False
+    )
+    df_poly_features.set_output(transform="pandas")
+    df_trans = df_poly_features.fit_transform(df_features)
+
+    # Create masked matrix with the interactions
+    n_weibull_parameters = 2 * n_events
+    w_star = rng.randn(df_trans.shape[1], n_weibull_parameters)
+    # Set 1-feature_rate% of the w_star to 0
+    w_star = np.where(
+        rng.rand(w_star.shape[0], w_star.shape[1]) < features_rate, w_star, 0
+    )
+    # Computation of the true values of shape and scale
+    shape_scale_star = df_trans.values @ w_star
+    shape_scale_columns = [f"shape_{i}" for i in range(1, n_events + 1)] + [
+        f"scale_{i}" for i in range(1, n_events + 1)
+    ]
+    df_shape_scale_star = pd.DataFrame(shape_scale_star, columns=shape_scale_columns)
+    # Rescaling of these values to stay in the chosen range
+    for event_id, shape_range, scale_range in zip(
+        range(1, n_events + 1), cycle(shape_ranges), cycle(scale_ranges)
+    ):
+        df_shape_scale_star = rescaling_params_to_respect_default_ranges(
+            df_shape_scale_star, shape_range, scale_range, event_id
+        )
+    return df_shape_scale_star
+
+
+def rescaling_params_to_respect_default_ranges(
+    df_shape_scale_star, shape_default, scale_default, event
+):
+    # Rescaling of these values to stay in the chosen range
+    shape_min, shape_max = shape_default
+    scale_min, scale_max = scale_default
+    shape = df_shape_scale_star[f"shape_{event}"].copy()
+    scale = df_shape_scale_star[f"scale_{event}"].copy()
+    scaler = MinMaxScaler(feature_range=(-2, 2))
+    df_shape_scale_star[f"shape_{event}"] = (
+        shape_min
+        + (shape_max - shape_min)
+        * expit(scaler.fit_transform(shape.values.reshape(-1, 1)))
```

Comment:
We don't need numpy conversion here, we can pass directly `df_shape_scale_star[[f"shape_{event}"]]` to the transformer.